test/Microsoft.ApplicationInsights.AspNet.Tests/Extensions/ApplicationInsightsExtensionsTests.cs

```diff
@@ -99,6 +82,130 @@ public static void RegistersTelemetryConfigurationFactoryMethodThatReadsInstrume
+                var config = new Configuration().AddEnvironmentVariables();
+                try
+                {
+                    services.AddApplicationInsightsTelemetry(config);
+
+                    IServiceProvider serviceProvider = services.BuildServiceProvider();
+                    var telemetryConfiguration = serviceProvider.GetRequiredService<TelemetryConfiguration>();
+                    Assert.Equal("11111111-2222-3333-4444-555555555555", telemetryConfiguration.InstrumentationKey);
+                }
+                finally
+                {
+                    Environment.SetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY", null);
+                }
+            }
+
+            [Fact]
+            public static void RegistersTelemetryConfigurationFactoryMethodThatReadsDeveloperModeFromEnvironment()
+            {
+                var services = ApplicationInsightsExtensionsTests.GetServiceCollectionWithContextAccessor();
+                Environment.SetEnvironmentVariable("APPINSIGHTS_DEVELOPER_MODE", "true");
+                var config = new Configuration().AddEnvironmentVariables();
+                try
+                {
+                    services.AddApplicationInsightsTelemetry(config);
+
+                    IServiceProvider serviceProvider = services.BuildServiceProvider();
+                    var telemetryConfiguration = serviceProvider.GetRequiredService<TelemetryConfiguration>();
+                    Assert.Equal(true, telemetryConfiguration.TelemetryChannel.DeveloperMode);
+                }
+                finally
+                {
+                    Environment.SetEnvironmentVariable("APPINSIGHTS_DEVELOPER_MODE", null);
+                }
+            }
+
+            [Fact]
+            public static void RegistersTelemetryConfigurationFactoryMethodThatReadsEndpointAddressFromEnvironment()
+            {
+                var services = ApplicationInsightsExtensionsTests.GetServiceCollectionWithContextAccessor();
+                Environment.SetEnvironmentVariable("APPINSIGHTS_ENDPOINTADDRESS", "http://localhost:1234/v2/track/");
+                var config = new Configuration().AddEnvironmentVariables();
+                try
+                {
+                    services.AddApplicationInsightsTelemetry(config);
+
+                    IServiceProvider serviceProvider = services.BuildServiceProvider();
+                    var telemetryConfiguration = serviceProvider.GetRequiredService<TelemetryConfiguration>();
+                    Assert.Equal("http://localhost:1234/v2/track/", ((InProcessTelemetryChannel)telemetryConfiguration.TelemetryChannel).EndpointAddress);
+                }
+                finally
+                {
+                    Environment.SetEnvironmentVariable("APPINSIGHTS_ENDPOINTADDRESS", null);
+                }
+            }
+
+            [Fact]
+            public static void RegistersTelemetryConfigurationFactoryMethodThatReadsInstrumentationKeyFromSettings()
```

Comment:
This and other "settings" related tests below verify the `AddApplicationInsightsSettings` method as opposed to the `AddApplicationInsightsTelemetry` method. Can we move them into a separate test class?